st2common/st2common/models/db/execution.py

```diff
@@ -110,6 +111,14 @@ def mask_secrets(self, value):
         if 'parameters' in liveaction:
             liveaction['parameters'] = mask_secret_parameters(parameters=liveaction['parameters'],
                                                               secret_parameters=secret_parameters)
+
+        # TODO(mierdin): This logic should be moved to the dedicated Inquiry
+        # data model once it exists.
+        if self.runner.get('name') == "inquirer":
+            response = result['result'].get('response')
+            schema = result['result'].get('schema')
```

Comment:
Should probably do `schema = schema or {}` or similar here or in `mask_inquiry_response` since it seems like now it would explode in case schema is `None`.

(on a related note - is it even possible for schema to be empty and / or not have `properties` attribute?)

Reply:
Very good point. I tested this and in fact you *can* create an Inquiry with an empty schema. This results in bad behavior at respond time that I documented in https://github.com/StackStorm/st2/issues/3846 - I'll make sure to fix that in an upcoming release.

In the meantime, I changed this line to do an explicit key lookup for `schema` rather than using `.get()`. Unlike the `response` key, the `schema` key will always be there (enforced at the `inquirer` runner level), and this code shouldn't care what the contents of the schema are, it should only pass it to `mask_inquiry_response`.

Reply:
Still not working - looking at this now

Reply:
Okay should be good now. In https://github.com/StackStorm/st2/pull/3825/commits/bf57136fc2d0dc4812a87fd1702beb6a864c4f19 I decided to revert back to `.get()` and default to an empty dict. In addition, I'm checking to ensure it's not empty before moving on to the masking function.